drivers/serial/uart_imx.c

```diff
@@ -113,16 +113,18 @@ static void uart_imx_poll_out(const struct device *dev, unsigned char c)
 static int uart_imx_poll_in(const struct device *dev, unsigned char *c)
 {
 	UART_Type *uart = UART_STRUCT(dev);
+	int ret = -1;
```

Comment:
Could we use the `-EIO` (`-5`) instead of `-EPERM` (`-1`)?

Reply:
Sorry, I didn't look into the UART header. Not sure how `-1` got in there, normally `-<ERRNO>` is used from `errno.h`.

Reply:
Please do add an issue and if possible a PR if this is something that needs to be addressed.